src/buildstream/_frontend/widget.py

```diff
@@ -461,6 +461,25 @@ def show_pipeline(self, dependencies, format_):
                 # Dump the SourceInfo provenance objects in yaml format
                 line = p.fmt_subst(line, "source-info", _yaml.roundtrip_dump_string(all_source_infos))
 
+            # Artifact CAS Digest
+            if "%{artifact-cas-digest" in format_:
+                artifact = element._get_artifact()
+                if not artifact.query_cache():
+                    artifact = None
+                if artifact is not None:
```

Comment:
This is bad style, just check the actual condition directly

```suggestion
                if artifact.query_cache():
```

Better yet, since we have already queried the cache, we just need to check whether the artifact is cached
```suggestion
                if artifact.cached():
```